logilica_weekly_report/page_settings.py

```diff
@@ -0,0 +1,210 @@
+            settings_category.wait_for(state="visible", timeout=self.DROPDOWN_TIMEOUT)
+
+        # open dashboard
+        settings_category.click()
+
+    def open_integration_configuration(self, integration: str, connector: str) -> None:
+        logging.debug(
+            "Opening integration '%s', connector type:'%s')", integration, connector
+        )
+        self.page.locator("div.items-center").filter(has_text=connector).filter(
+            has_text=integration
+        ).get_by_role("button", name="Configure").click()
+        expect(
+            self.page.get_by_role("heading", name=f"{connector} Settings")
+        ).to_be_visible()
+
+    def sync_integrations(self, integrations: dict[str, Any]) -> None:
+
+        sync_failures: IntegrationSyncFailures = {}
+        for integration_name, details in integrations.items():
+            connector = details["connector"]
+            public_repos = details.get("public_repositories", [])
+            member_repos = details.get("membership_repositories", [])
+            logging.debug(
+                "Syncing integration '%s' (connector '%s') , %d public repositories, %d membership based repositories",
+                integration_name,
+                connector,
+                len(public_repos),
+                len(member_repos),
+            )
+
+            # open configuration UI
+            self.open_settings("Integrations")
+            self.open_integration_configuration(
+                integration=integration_name, connector=connector
+            )
+
+            # process public repositories
+            added_repositories: list[str] = []
+            for repo_slug in public_repos:
+                if not self.has_repository_imported(repo_slug):
+                    self.add_public_repository(repo_slug)
+                    added_repositories.append(repo_slug)
+            missing_repositories = self.check_imported_repositories(added_repositories)
+            if missing_repositories:
+                sync_failures[(connector, integration_name)] = missing_repositories
+
+            # process membership repositories
+            if len(member_repos) > 0:
+                logging.debug(
+                    "Waiting up to %d milliseconds for membership repositories to appear on the screen prior starting adding them.",
+                    self.AVAILABLE_REPOSITORY_TIMEOUT,
+                )
+                self.page.wait_for_timeout(self.AVAILABLE_REPOSITORY_TIMEOUT)
+
+                added_repositories = []
+                for repo_slug in member_repos:
+                    if not self.has_repository_imported(repo_slug):
+                        if self.add_membership_repository(repo_slug):
+                            added_repositories.append(repo_slug)
+                        else:
+                            sync_failures.setdefault(
+                                (connector, integration_name), []
+                            ).append(
```

Comment:
This is clever but really ugly with all the line breaks, so I suggest using a [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) instead.